sklearn/metrics/_classification.py

```diff
@@ -3556,10 +3605,12 @@ def _validate_binary_probabilistic_prediction(y_true, y_prob, sample_weight, pos
         else:
             raise
 
+    transformed_labels = _one_hot_encoding_binary_target(
+        y_true=y_true, pos_label=pos_label, target_xp=xp, target_device=device_
+    )
     # convert (n_samples,) to (n_samples, 2) shape
-    y_true = np.array(y_true == pos_label, int)
-    transformed_labels = np.column_stack((1 - y_true, y_true))
-    y_prob = np.column_stack((1 - y_prob, y_prob))
+    transformed_labels = xp.stack((1 - transformed_labels, transformed_labels), axis=1)
```

Comment:
I think it would make sense to move this line into `_one_hot_encoding_binary_target`.

Reply:
Then maybe we can also move a similar one in the multiclass case too?